### DISCUSSION TITLE
Carbon Design CUSTOM tokens ERROR, Unable to find token!

### DISCUSSION THREAD

**Discussion by DenisCor** [carbon-design-system/carbon#12908] [Help]
https://github.com/carbon-design-system/carbon/blob/main/examples/custom-theme/src/index.scss followed this example to the teeth plus few others and nothing seem to work. 

```
@use '@carbon/react/scss/themes' as *;
@use '@carbon/react/scss/theme' with (
  $theme: (text-primary: #a235aa, // <-- text-primary is a Carbon token
   custom-token: #3f51b5) // <-- my custom token I want to use 
);
```

say I want to use these tokens in other scss files, I add `@use '@carbon/react/scss/theme';` to the top

Example:
```
 .some-class {
color: theme.$text-primary;  // <-- all carbon default tokens work great, no problem here
}
```
BUT if I want to use my custom-token I get an error

Example: 

```
.some-class {
color: theme.get('custom-token');  // <-- ERROR: 
}
```

Module build failed (from ./node_modules.. etc etc)
SassError: "unable to find token: custom-token in current $theme.

**Comment by tay1orjones** [comment]
@DenisCor Are you bringing in the second file via `@use`? [Here's an example of it working](https://stackblitz.com/edit/github-i3mhsa?file=src%2Findex.scss,src%2Ftest.scss).

If you're importing the second file via javascript, sass will be executed against the files individually and they will not share an execution context.

**Reply by DenisCor** [reply]
this wont work for me!  did exactly what you have in your example and ill still get the "unable to find token: custom-token in current $theme" Im suspecting that in your example you used vite.

**Reply by tay1orjones** [reply]
> Im suspecting that in your example you used vite.

Yes. What are you using? If webpack, what version of `webpack` (or `react-scripts`), `sass`, and `sass-loader` are you using?

**Reply by pakeecbala** [reply]
I have the same issue. Im using Next.js 13.4.8 and sass 1.51. Any solution?

**Reply by tay1orjones** [reply]
Could you provide a link to a reproduction? It's hard to know what's going wrong without seeing it firsthand

**Reply by pakeecbala** [reply]
Sure, here you go :)

https://stackblitz.com/edit/stackblitz-starters-xwnxzu?file=app%2Fabout%2Fabout.module.scss

**Reply by DenisCor** [reply]
use this instead - `@use '@carbon/styles/scss/utilities/custom-property' as theme`; then in your styles -  `background-color: theme.get-var('your-custom-token-name');`
